File: app/src/main/java/dev/yiray/qickmvivm/base/IViewModel.java
package dev.yiray.qickmvivm.base;

import androidx.lifecycle.LiveData;

public interface IViewModel<S> {
    LiveData<S> states();
    LiveData<Throwable> errors();
    LiveData<SideEffectEvent<S>> sideEffects();
    void handleSideEffects(S state, S sideEffect);
}

File: app/src/main/java/dev/yiray/livedatamvi/base/IViewModel.kt
package dev.yiray.livedatamvi.base

import androidx.lifecycle.LiveData

interface IViewModel<S> {
    fun states(): LiveData<S>
    fun errors(): LiveData<Throwable?>
    fun sideEffects(): LiveData<SideEffectEvent<S>>
    fun handleSideEffects(state: S, sideEffect: S)
}